test/core.spec.ts

```diff
@@ -896,6 +896,130 @@ contract("Core", (accounts) => {
     });
   });
 
+  describe("#redeem", async () => {
+    let subjectCaller: Address;
+    let subjectQuantityToRedeem: BigNumber;
+    let subjectSetToRedeem: Address;
+
+    const naturalUnit: BigNumber = ether(2);
+    let components: StandardTokenMockContract[] = [];
+    let componentUnits: BigNumber[];
+    let setToken: SetTokenContract;
+
+    beforeEach(async () => {
+      await deployCoreAndInitializeDependencies();
+
+      components = await coreWrapper.deployTokensAsync(2, ownerAccount);
+      await coreWrapper.approveTransfersAsync(components, transferProxy.address);
+
+      const componentAddresses = _.map(components, (token) => token.address);
+      componentUnits = _.map(components, () => ether(4)); // Multiple of naturalUnit
+      setToken = await coreWrapper.createSetTokenAsync(
+        core,
+        setTokenFactory.address,
+        componentAddresses,
+        componentUnits,
+        naturalUnit,
+        "Set Token",
+        "SET",
+      );
+
+      await coreWrapper.issueSetTokenAsync(
+        core,
+        setToken.address,
+        ether(2),
+      );
+
+      subjectCaller = ownerAccount;
+      subjectQuantityToRedeem = ether(2);
+      subjectSetToRedeem = setToken.address;
+    });
+
+    async function subject(): Promise<string> {
+      return core.redeem.sendTransactionAsync(
+        subjectSetToRedeem,
+        subjectQuantityToRedeem,
+        { from: ownerAccount },
+      );
+    }
+
```

Comment:
Add a test for decremented `setToken` balance of the user, because it got burned

Reply:
I added that in the `setToken.spec.ts`

Reply:
Hmm is there a way to validate that burn still got called though at the right place?

Reply:
added spec